### ISSUE TITLE
xq's fix to enum_map.hpp

### ISSUE THREAD

**Issue by clort81** [calref/cboe#203]
```
diff cboe/src/tools/enum_map.hpp cboe-dark/src/tools/enum_map.hpp
16a17,23
> 
> // clort added by xq 
>     using typename std::array<T, n>::reference;
>     using typename std::array<T, n>::const_reference;
>     using typename std::array<T, n>::size_type;
>     using std::array<T, n>::begin;
> 
25,27c32,40
< 	reference at(E pos) {return array::at(static_cast<int>(pos));};
< 	const_reference at(E pos) const {return array::operator[](static_cast<int>(pos));};
< 	reference operator[](E pos) {return array::operator[](static_cast<int>(pos));};
---
> 
>         // clort added by xq
> 	//reference at(E pos) {return array::at(static_cast<int>(pos));};
> 	//const_reference at(E pos) const {return array::operator[](static_cast<int>(pos));};
> 	//reference operator[](E pos) {return array::operator[](static_cast<int>(pos));};
>         reference at(E pos) {return std::array<T, n>::at(static_cast<int>(pos));};
>         const_reference at(E pos) const {return std::array<T, n>::operator[](static_cast<int>(pos));};
>         reference operator[](E pos) {return std::array<T, n>::operator[](static_cast<int>(pos));};
> 
```

this fixes building for me on linux.  cheers.

**Comment by CelticMinstrel** [comment]
Just noting: I'd also add `using std::array<T, n>::end` for symmetry... also, probably add a type alias for `std::array<T, n>` to simplify all of these.

**Comment by madwizard** [comment]
Fixes build on Fedora 31.

**Comment by CelticMinstrel** [comment]
Fixed by e42baa0971f3a231b5eb76c98b556fc4a0635dc2 fa25beae6d66456d7507ffec34252f58a3ad092f fb28a4da2e817c61abcb19c81d28e6b73511a92a
